core/src/main/java/org/apache/accumulo/core/iterators/IteratorEnvironment.java

```diff
@@ -86,12 +124,52 @@ default boolean isUserCompaction() {
    * @return true if sampling is enabled for this environment.
    * @since 1.8.0
    */
-  boolean isSamplingEnabled();
+  default boolean isSamplingEnabled() {
+    throw new UnsupportedOperationException();
+  }
 
   /**
    *
    * @return sampling configuration is sampling is enabled for environment, otherwise returns null.
    * @since 1.8.0
    */
-  SamplerConfiguration getSamplerConfiguration();
+  default SamplerConfiguration getSamplerConfiguration() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * True if compaction was user initiated.
+   *
+   * @since 2.0.0
+   */
+  default boolean isUserCompaction() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns an object containing information about the server where this iterator was run.
+   *
+   * @since 2.0.0
+   */
+  default ServiceEnvironment getServiceEnv() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns an object containing the table configuration for this iterator.
+   *
+   * @since 2.0.0
+   */
+  default ServiceEnvironment.Configuration getTableConfiguration() {
```

Comment:
I would just add javadoc to getConfiguration showing how this could be obtained if needed rather than create a duplicate way of doing this in the api

Reply:
I will do that.  I added getTableConfiguration() to the interface for convenience but it does seem unnecessary.

Reply:
I don't think the IteratorEnvironment is duplicating it... it's just wrapping what it has available to provide it's own API. Since table configuration is very commonly needed within an iterator, and an iterator runs within the scope of a single specific table, I think this is worth keeping in the IteratorEnvironment, so users don't have to jump through hoops to reach down into the ServerEnvironment.

Reply:
We can always add it later if a user requests it.  I lean on less is better in the API since its much easier to add something then remove it.

Reply:
That's fine with me.